lib/kaguya/core.ex

```diff
@@ -49,21 +50,31 @@ defmodule Kaguya.Core do
   end
 
   def handle_info({:tcp, _socket, messages}, state) do
+    state = server_timer(state, server_timeout())
     for msg <- String.split(String.rstrip(messages), "\r\n"), do: handle_message(msg)
     {:noreply, state}
   end
 
   def handle_info({:ssl, _socket, messages}, state) do
+    state = server_timer(state, server_timeout())
     for msg <- String.split(String.rstrip(messages), "\r\n"), do: handle_message(msg)
     {:noreply, state}
   end
 
-  def handle_info({:tcp_closed, _port}, _state) do
+  def handle_info({:tcp_closed, _port}, state) do
+    cancel_server_timer(state)
     socket = reconnect()
     {:noreply, %{socket: socket}}
```

Comment:
It seems incorrect that the returned state here isn't used to continue the timer once this function returns. That is, if the tcp connection is closed, then reconnected won't the timer never recreate itself since state will lack the :server_timer attribute?